packages/tizen_app_manager/tizen/src/application_utils.cc

```diff
@@ -4,145 +4,194 @@
+              static_cast<flutter::EncodableMap *>(user_data);
+          metadata->insert(
+              std::pair<flutter::EncodableValue, flutter::EncodableValue>(
+                  std::string(key), std::string(value)));
+          LOG_INFO("AppMetaDataCB key: %s, value: %s", key, value);
+        }
+        return true;
+      },
+      (void *)&metadata);
   if (ret != APP_MANAGER_ERROR_NONE) {
     LOG_ERROR("app_info_foreach_metadata error! : %s", get_error_message(ret));
-    goto cleanup;
-  }
-
-  value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
-      "appId", std::string(app_id)));
-  value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
-      "packageId", std::string(pkg_id)));
-  value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
-      "label", std::string(label)));
-  value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
-      "type", std::string(type)));
-  if (icon_path != nullptr) {
-    value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
-        "iconPath", std::string(icon_path)));
+    throw ret;
   }
-  value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
-      "executablePath", std::string(exec_path)));
-  value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
-      "sharedResourcePath", std::string(shared_res_path)));
-  value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
-      "isNoDisplay", no_display));
   value.insert(std::pair<flutter::EncodableValue, flutter::EncodableMap>(
       "metadata", metadata));
+}
+}  // namespace
 
-cleanup:
-  if (app_id) {
-    free(app_id);
-  }
-  if (pkg_id) {
-    free(pkg_id);
-  }
-  if (label) {
-    free(label);
-  }
-  if (type) {
-    free(type);
-  }
-  if (icon_path) {
-    free(icon_path);
+namespace application_utils {
+bool ExtractValueFromMap(const flutter::EncodableValue &arguments,
+                         const char *key, std::string &out_value) {
+  if (std::holds_alternative<flutter::EncodableMap>(arguments)) {
+    flutter::EncodableMap map = std::get<flutter::EncodableMap>(arguments);
+    auto iter = map.find(flutter::EncodableValue(key));
+    if (iter != map.end() && !iter->second.IsNull()) {
+      if (auto pval = std::get_if<std::string>(&iter->second)) {
+        out_value = *pval;
+        return true;
+      }
+    }
   }
-  if (exec_path) {
-    free(exec_path);
+  return false;
+}
+
+int GetAppData(app_info_h app_info, flutter::EncodableMap &value) {
+  char *app_id = nullptr;
+  int ret = APP_MANAGER_ERROR_NONE;
+  try {
+    app_id = GetAppId(app_info, value);
+    GetPkgId(app_info, value);
+    GetLabel(app_info, value);
+    GetType(app_info, value);
+    GetIconPath(app_info, value);
+    GetExecutablePath(app_info, value);
+    GetSharedResourcePath(app_id, value);
+    GetIsNoDisplay(app_info, value);
+    GetForEachMetadata(app_info, value);
```

Comment:
Why don't you make `AppInfo` into a class?

Reply:
Created `AppData` class. This class supports `GetAppData ()` and stores the app_info and value mentioned as member variables.
It seems to have been modified in a different direction from the direction you reviewed. Please review the code again to see if the current code contains the intention of the review.

Reply:
1. `AppData` → `AppInfo`
2. Move the class into a separate file.
3. Remove the `GetAppData` method and the private member `flutter_map_`. You'll need to refactor the code a bit.
4. Do not use C++ exceptions.
5. Do not abbreviate words: `GetPkgId` → `GetPackageId`.

Please ask me or @bbrto21 offline if you need a rationale or detailed explanation.

Reply:
I applied the reviews mentioned.
1. `AppData`-> `TizenAppInfo`
2. The defined class is separated into a separate file. Tizen_app_info.h was created to distinguish from App_info.h, and the prefix Tizen was added to the class name.
3. Private Member `Flutter_map_` was removed from the class. It seems to be inappropriate because it was a private member, but it could be affected outside of the class.
In the review, you requested to remove the `GetAppdata` function, but we have made a modification.
4. The C ++ Exceptions I used was removed.
5. The part where the compression was used is modified.